File: 2022/9/24/1.LeetCode1652.cpp
/*************************************************************************
	> File Name: 1.LeetCode1652.cpp
	> Author: 
	> Mail: 
	> Created Time: 2022年09月24日 星期六 09时33分58秒
 ************************************************************************/

#include <iostream>
#include <string>
#include <vector>
#include <queue>
#include <stack>
#include <map>
#include <set>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
using namespace std;

int wzmw_zr = 0;
bool fuck_plagiarism_system_of_leetcode = true;

vector<int> decrypt(vector<int>& code, int k) {
  int n = code.size();
  vector<int> ans(n, 0);
  if (k == 0) {
    return ans;
  } else if (k > 0) {
    for (int i = 0; i < n; i++) {
      int sum = 0;
      for (int j = 1; j <= k; j++) 
        sum += code[(i + j) % n];
      ans[i] = sum;
    }
  } else {
    for (int i = 0; i < n; i++) {
      int sum = 0;
      for (int j = -1; j >= k; j--) 
        sum += code[(i + j + n) % n];
      ans[i] = sum;
    }
  }
  return ans;
}

int main() {
  return 0;
}

File: 2024/5/5/1.LeetCode1652.go
package main

import (
  "fmt"
)

func decrypt(code []int, k int) []int {
  var n int = len(code)
  var ans []int = make([]int, n)
  var flag int
  if k > 0 {
    flag = 1
  } else {
    flag = -1
  }
  k *= flag
  var sum int
  for i := 1; i <= k; i++ {
    var idx int = (flag * i + n) % n
    sum += code[idx]
  }
  for i := 0; i < n; i++ {
    ans[i] = sum
    if flag > 0 {
      sum = sum - code[(i + 1) % n] + code[(i + k + 1) % n]
    } else {
      sum = sum - code[(i - k + n) % n] + code[i]
    }
  }
  return ans
}

func main() {
  var code []int = []int{2, 4, 9, 3}
  var k int = -2
  fmt.Println(decrypt(code, k))
}
